aws-amplify-publisher-plugin/src/utils.tsx

```diff
@@ -50,52 +49,17 @@ async function zipFile(
 
 /**
  * Compress the files needed to be host on the server.
- * @param folderPrefix the prefix of the folder's name.
  * @returns A Promise object that resolves the zip file path.
- * @throws An error if it fails to get WorkSpace DirPath or it does not exist.
  */
-export async function zipArtifacts(folderPrefix: string): Promise<string> {
-  let tmpDir: string | undefined;
+export async function zipArtifacts(): Promise<string> {
   const workSpaceDirPath = WorkSpace.DirPath;
-  try {
-    if (workSpaceDirPath !== null && fs.existsSync(workSpaceDirPath)) {
-      const now = new Date();
-      tmpDir = fs.mkdtempSync(
-        path.join(workSpaceDirPath, `${folderPrefix}${now.getTime()}`)
-      );
-
-      // get the scenes folder from the Editor project directory
-      const tmpDirScenes = path.join(tmpDir, 'scenes');
-      const scenesFolder = path.join(workSpaceDirPath, 'scenes');
-      fs.copySync(scenesFolder, tmpDirScenes);
-
-      // get the dist folder from the Editor project directory
-      const tmpDirDist = path.join(tmpDir, 'dist');
-      const distFolder = path.join(workSpaceDirPath, 'dist');
-      fs.copySync(distFolder, tmpDirDist);
-
-      // get the index.html from the Editor project directory
-      const tempIndexHTML = path.join(tmpDir, 'index.html');
-      const indexHTML = path.join(workSpaceDirPath, 'index.html');
-      fs.copyFileSync(indexHTML, tempIndexHTML);
-    } else {
-      throw new Error(
-        `Cannot get WorkSpace directory ${workSpaceDirPath}, or it does not exist.`
-      );
-    }
-    const zipFilePath = `${tmpDir}.zip`;
-    return await zipFile(tmpDir, zipFilePath);
-  } finally {
-    try {
-      if (tmpDir) {
-        fs.rmdirSync(tmpDir, {recursive: true});
-      }
-    } catch (error) {
-      console.error(
-        `An error has occurred while removing the temp folder at ${tmpDir}. Please remove it manually. ${error}`
-      );
-    }
+  if (workSpaceDirPath == null || !fs.existsSync(workSpaceDirPath)) {
+    throw new Error(
+      `Cannot get WorkSpace directory ${workSpaceDirPath}, or it does not exist.`
+    );
   }
+  const zipFilePath = `${workSpaceDirPath}.zip`;
```

Comment:
This code seems far less complex - for my understanding, are there drawbacks in zipping up the whole project directory, versus selectively zipping up specific files the way you were before? 

One pro of doing it this way is you're entirely less dependent on changes to the project directory structure, but I suppose there's some risk in asset duplication / archive sizes?

Reply:
The con is taking more time to publish since after this change, more files need to be loaded to Amplify side.